assets/js/components/setup/setup-proxy.js

```diff
@@ -42,177 +41,135 @@ import Notification from '../notifications/notification';
+	} = useSelect( ( select ) => {
+		const site = select( CORE_SITE );
+		const user = select( CORE_USER );
+
+		return {
+			isSecondAdmin: site.hasConnectedAdmins(),
+			isResettable: site.isResettable(),
+			siteURL: site.getReferenceSiteURL(),
+			proxySetupURL: site.getProxySetupURL(),
+			disconnectedReason: user.getDisconnectedReason(),
 		};
+	} );
+
+	const onButtonClick = useCallback( async ( event ) => {
+		event.preventDefault();
+		await trackEvent( 'plugin_setup', 'proxy_start_setup_landing_page' );
+		global.location.assign( proxySetupURL );
+	}, [ proxySetupURL ] );
+
+	// @TODO: this needs to be migrated to the core/site datastore in the future
+	const { errorMessage } = global._googlesitekitLegacyData.setup;
+
+	let title;
+	let description;
+	let startSetupText;
+
+	if ( 'revoked' === getQueryArg( location.href, 'googlesitekit_context' ) ) {
+		title = sprintf(
+			/* translators: %s is the site's hostname. (e.g. example.com) */
+			__( 'You revoked access to Site Kit for %s', 'google-site-kit' ),
+			punycode.toUnicode( ( new URL( siteURL ) ).hostname )
+		);
+		description = __( 'Site Kit will no longer have access to your account. If you’d like to reconnect Site Kit, click "Start Setup" below to generate new credentials.', 'google-site-kit' );
+		startSetupText = __( 'Sign in with Google', 'google-site-kit' );
+	} else if ( isSecondAdmin ) {
+		title = __( 'Connect your Google account to Site Kit', 'google-site-kit' );
+		description = __( 'Site Kit has already been configured by another admin of this site. To use Site Kit as well, sign in with your Google account which has access to Google services for this site (e.g. Google Analytics). Once you complete the 3 setup steps, you’ll see stats from all activated Google products.', 'google-site-kit' );
+		startSetupText = __( 'Sign in with Google', 'google-site-kit' );
+	} else if ( DISCONNECTED_REASON_CONNECTED_URL_MISMATCH === disconnectedReason ) {
+		title = __( 'Reconnect Site Kit', 'google-site-kit' );
+		description = __( `Looks like the URL of your site has changed. In order to continue using Site Kit, you'll need to reconnect, so that your plugin settings are updated with the new URL.`, 'google-site-kit' );
+		startSetupText = __( 'Reconnect', 'google-site-kit' );
+	} else {
+		title = __( 'Sign in with Google to set up Site Kit', 'google-site-kit' );
+		description = __( 'The Site Kit service will guide you through 3 simple setup steps.', 'google-site-kit' );
+		startSetupText = __( 'Start setup', 'google-site-kit' );
 	}
 
-	isSetupFinished() {
-		const { completeSetup } = this.state;
-		const { isConnected } = this.props;
-
-		return isConnected && completeSetup;
-	}
-
-	render() {
-		if ( this.isSetupFinished() ) {
-			const redirectURL = getSiteKitAdminURL(
-				'googlesitekit-dashboard',
-				{
-					notification: 'authentication_success',
-				},
-			);
-
-			delay( () => {
-				global.location.replace( redirectURL );
-			}, 500, 'later' );
-		}
```

Comment:
I removed it because it was dead code. The `completeSetup` variable always had `false` value, hence `this.isSetupFinished()` always returned `false` and `if` condition couldn't be reached.